### ISSUE TITLE
Proposal: Default Casting of Integers to Long Types

### ISSUE THREAD

**Issue by dan1elt0m** [mitchelllisle/sparkdantic#351]
**Feature Description**

I propose that Sparkdantic should by default cast integers to Spark's LongType instead of IntegerType. This change would help prevent issues with integer overflow and make the library more robust for handling large numbers.

**Proposed Solution**

Introduce an option called `safe_casting` that, when enabled, would cast integers to LongType. The default value of this option could be `True` to ensure safety by default.

**Example**

Here's an example of how this feature could be used:

```python
from pyspark.sql import SparkSession
from sparkdantic import SparkModel

class MyModel(SparkModel):
    my_field: int

# With safe_casting enabled, this would cast 'my_field' to LongType
model = MyModel.spark_schema(safe_casting=True)
```

Look forward to hearing your thoughts. I am also willing to create a PR for this feature.

**Comment by mitchelllisle** [comment]
Hi @dan1elt0m ,
Thanks for all the contributions and suggestions. I am not opposed to this approach - although this does sound like a breaking change to me since a number of people might rely on the casting of `int` to `IntegerType`. Does the ability to override the type also not work well enough for your use cases? I'd be keen to understand your specific use cases where the overflow is a problem

**Comment by dan1elt0m** [comment]
Hey @mitchelllisle, 

Thank you for your feedback and for creating this great library. I understand your concern regarding the potential for this change to be breaking for users who rely on the current casting behaviour.

For my usecase, Sparkdantic generates the schema's of many Databricks catalog tables and also generates the schema's for casting dataframes to the required structure before writing to the Databricks catalog. In the catalog, there tens of tables with easily over 50 columns each, which are often nested. Handling these large nested structures by manually setting the overrides to ensure the correct data type casting is impractical and error-prone.  Automating the cast to LongType by default simplifies the workflow, while ensuring data integrity across all tables.

To address the breaking change concern, the safe_casting option could be introduced in a way that maintains backward compatibility. For example, setting safe_casting=True as the default while allowing users to explicitly set safe_casting=False if they need to retain the current IntegerType casting. Ofcourse, this needs to be documented well.

**Comment by mitchelllisle** [comment]
How would you feel about a default of `safe_casting=False`?

**Comment by dan1elt0m** [comment]
I can live with that

**Comment by mitchelllisle** [comment]
Awesome. We can revisit when it feels right to move to 1.0.0

**Comment by dan1elt0m** [comment]
https://github.com/mitchelllisle/sparkdantic/pull/367

**Comment by mitchelllisle** [comment]
Released in [v0.25.0](https://github.com/mitchelllisle/sparkdantic/releases/tag/v0.25.0)
